CAP/gap/ToolsForCategories.gi

```diff
@@ -686,7 +686,7 @@ InstallGlobalFunction( CAP_INTERNAL_GET_CORRESPONDING_OUTPUT_OBJECTS,
             if ForAll( current_output[ 2 ], i -> i in "0123456789" ) then
                 list_position := Int( current_output[ 2 ] );
             else
-                list_position := Position( input_list, current_output[ 2 ] );
+                list_position := function_input[ Position( input_list, current_output[ 2 ] ) ];
```

Comment:
If `Position` returns `fail` then we don't have a meaningful error message anymore.
I suggest to write
```
list_position := function_input[ list_position ];
```
right after the `list_position = fail` statement, i.e., in line 695.

Reply:
There are two assignments of `list_position` in the `if-else` statement. The first one is correct and the second was buggy.

Your suggestion will also alter the first correct assignment and break it.

Reply:
Yes, you are right. But this is easy to overcome. Could you please fix it?

Reply:
I added a commit with an error message.